File: Leetcode-cpp/SameTree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode *p, TreeNode *q) {
       if(p == NULL || q == NULL) return p == NULL && q == NULL;
       else return p->val == q->val && isSameTree(p->left,q->left) && isSameTree(p->right,q->right);
    }
};

File: LeetCode-Java/Same Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        return (p == null || q == null) ? (p == null && q == null) : ((p.val == q.val) && isSameTree(p.left,q.left) && isSameTree(p.right, q.right));
    }
}